packages/blade/src/components/IconButton/StyledIconButton.web.tsx

```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+
+type StyledClickableProps = {
+  contrast: 'low' | 'high';
+};
+
+const StyledClickable = styled.button<StyledClickableProps>((props) => {
+  const { theme, contrast } = props;
+  const linkColorToken = theme.colors.feedback.neutral.action.icon.link; // todo: should be from surfaceAction
+  const contrastToken = contrast === 'high' ? 'highContrast' : 'lowContrast';
+  const motionToken = theme.motion;
+
+  return {
+    border: 'none',
+    cursor: 'pointer',
+    padding: 0,
+    borderRadius: theme.border.radius.small,
+    background: 'transparent',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    color: linkColorToken.default[contrastToken],
+    transition: `all ${motionToken.duration.xquick} ${motionToken.easing.standard.effective}`,
+
+    'svg, path': {
+      // todo: should be ideally set in Icon component
+      fill: 'currentColor',
```

Comment:
How does currentColor work in react-native?

Reply:
[Support](https://github.com/react-native-svg/react-native-svg/blob/main/USAGE.md#svg) is there. Though I haven't implemented the native code *yet*. Should work in theory 🤞

Reply:
React native doesn't support `&:hover`, `&:active` so even if we use `currentColor` there, we'll have to change the color with Pressable's `pressed` from it's style or children (similar to the implementation on Link). 

This approach looks cleaner for web though, we can avoid JS here 👌

Reply:
So it seems with `react-native-svg` it can only derive the `currentColor` on elements inside `Svg`, not directly on `Svg`, having `color="currentColor"` crashes it. So it seems for native we'll have to use similar approach we did earlier with Animated and Pressable.